test/common/http/conn_manager_impl_test_2.cc

```diff
@@ -520,8 +522,53 @@ TEST_F(HttpConnectionManagerImplTest, ConnectionDuration) {
   EXPECT_EQ(1U, stats_.named_.downstream_cx_max_duration_reached_.value());
 }
 
+TEST_F(HttpConnectionManagerImplTest, ConnectionDurationSafeHttp1) {
```

Comment:
Consider having an integration test demonstrating the client needs to initiate the close e.g. we can do multiple req<->resp after the initial close from the server.